Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -1530,6 +1548,38 @@ void FitPeaks::calculateFittedPeaks(const std::vector<std::shared_ptr<FitPeaksAl
 
   return;
 }
+
+double FitPeaks::calculateSignalToSigmaRatio(const size_t &iws, const std::pair<double, double> &peakWindow,
+                                             const API::IPeakFunction_sptr &peakFunction,
+                                             const API::IBackgroundFunction_sptr &backgroundFunction) {
+  const auto &vecX = m_fittedPeakWS->points(iws);
+  auto startX = std::lower_bound(vecX.begin(), vecX.end(), peakWindow.first);
+  auto stopX = std::lower_bound(vecX.begin(), vecX.end(), peakWindow.second);
+
+  FunctionDomain1DVector domain(startX, stopX);
+  FunctionValues values(domain);
+  CompositeFunction_sptr compFunc = std::make_shared<API::CompositeFunction>();
+
+  compFunc->addFunction(backgroundFunction);
+  compFunc->function(domain, values);
+  auto bkgdValues = values.toVector();
+
+  compFunc->addFunction(peakFunction);
+  compFunc->function(domain, values);
+  auto fittedValues = values.toVector();
+
+  const auto &errors = m_fittedPeakWS->readE(iws);
+  auto startE = std::lower_bound(errors.begin(), errors.end(), peakWindow.first);
+  auto stopE = std::lower_bound(errors.begin(), errors.end(), peakWindow.second);
+  std::vector<double> peakErrors(startE, stopE);
+
+  double fittedSum = std::accumulate(fittedValues.cbegin(), fittedValues.cend(), 0.0);
+  double bkgdSum = std::accumulate(bkgdValues.cbegin(), bkgdValues.cend(), 0.0);
+  double sigma = sqrt(std::accumulate(peakErrors.cbegin(), peakErrors.cend(), 0.0, VectorHelper::SumSquares<double>()));
+
+  return (fittedSum - bkgdSum) / ((sigma == 0) ? 1 : sigma);
```

Comment:
Actually just spotted that we could just evaluate the sum of the peak function evaluated  (with no background - i.e don't use a composite function)? Do you agree?

Reply:
So here we are just interested in the area under the peak - you can integrate the combined (background + peak) and subtract the integrated background, but because you have access to the whole composite function, you can choose instead just to integrate the peak part `integral(pk + bg) - integral(bg) = integral(pk) + integral(bg) - integral(bg) = integral(pk)`.
In previous discussions I did ask for you to subtract the background because I anticipated integrating the raw data not the fitted curve, but I think integrating the fitted curve is a similarly good check so happy to go with it!

Reply:
Oh, that makes sense. I don't know why I just assumed the integration is not linear. I think your simplification is better, so I will update the function to just calculate the peak function.